### ISSUE TITLE
Do-all and reduction patterns: Local variables classified as private

### ISSUE THREAD

**Issue by lukasrothenberger** [discopop-project/discopop#227]
Local variables are classified as `private` and wrongfully occur in the suggested pragma.

### Example 1:
* [x] Fixed
Might be fixed by #229.

```
for(int x = 0; x < 100; x++){
        int y = x;
}
```

Results in:
```
#pragma omp parallel for private(x,y)
for(int x = 0; x < 100; x++){
        int y = x;
}
```

Expected:
```
#pragma omp parallel for
for(int x = 0; x < 100; x++){
        int y = x;
}
```

### Example 2
* [x] Fixed

```
int z;
for(z = 0; z < 100; z++){
    int y = z;
}
```

Results in:
```
int z;
#pragma omp parallel for private(z,y)
for(z = 0; z < 100; z++){
    int y = z;
}
```

Expected:
```
int z;
#pragma omp parallel for private(z)
for(z = 0; z < 100; z++){
    int y = z;
}
```

**Comment by lukasrothenberger** [comment]
partially fixed in bc7e51f516f6e97c6e2224afcc284414d695b433.

**Comment by lukasrothenberger** [comment]
fixes #229.
